From afbe67f102551917b5c6edbf00ea442ac103ecf0 Mon Sep 17 00:00:00 2001
From: Alexandre Rostovtsev <tetromino@gmail.com>
Date: Fri, 2 Sep 2011 02:08:05 -0400
Subject: [PATCH] Do not assert if hostnamed is not running

Most Gentoo users do not use systemd; asserting if systemd's hostnamed
is not running is crazy.
---
 panels/info/cc-info-panel.c |   21 +++++++++++----------
 1 files changed, 11 insertions(+), 10 deletions(-)

diff --git a/panels/info/cc-info-panel.c b/panels/info/cc-info-panel.c
index 9e72e2a..0cb9342 100644
--- a/panels/info/cc-info-panel.c
+++ b/panels/info/cc-info-panel.c
@@ -1118,14 +1118,6 @@ info_panel_setup_hostname (CcInfoPanel  *self,
   char *str;
   GtkWidget *entry;
 
-  entry = WID ("name_entry");
-
-  gtk_widget_show (WID ("label4"));
-  gtk_widget_show (entry);
-
-  if (g_permission_get_allowed (permission) != FALSE)
-    gtk_widget_set_sensitive (entry, TRUE);
-
   self->priv->hostnamed_proxy = g_dbus_proxy_new_for_bus_sync (G_BUS_TYPE_SYSTEM,
                                                                G_DBUS_PROXY_FLAGS_NONE,
                                                                NULL,
@@ -1136,8 +1128,17 @@ info_panel_setup_hostname (CcInfoPanel  *self,
                                                                NULL);
 
   /* This could only happen if the policy file was installed
-   * but not hostnamed, which points to a system bug */
-  g_assert (self->priv->hostnamed_proxy);
+   * but not hostnamed */
+  if (!self->priv->hostnamed_proxy)
+    return;
+
+  entry = WID ("name_entry");
+
+  gtk_widget_show (WID ("label4"));
+  gtk_widget_show (entry);
+
+  if (g_permission_get_allowed (permission) != FALSE)
+    gtk_widget_set_sensitive (entry, TRUE);
 
   str = info_panel_get_hostname (self);
   gtk_entry_set_text (GTK_ENTRY (entry), str);
-- 
1.7.6.1

